ext/fg/js/popup.js

```diff
@@ -133,25 +133,15 @@ class Popup {
     }
 
     hide() {
-        this.hideContainer();
-        this.container.blur();
         this.hideChildren();
+        this.hideContainer();
+        this.focusParent();
     }
 
     hideChildren() {
-        if (this.children.length === 0) {
-            return;
-        }
-
-        const targets = this.children.slice(0);
-        while (targets.length > 0) {
-            const target = targets.shift();
-            if (target.isContainerHidden()) { continue; }
-
-            target.hideContainer();
-            for (const child of target.children) {
-                targets.push(child);
-            }
+        // recursively hides all children
+        if (this.child && !this.child.isContainerHidden()) {
+            this.child.hide();
```

Comment:
Is it necessary for these ```child.hide``` calls to invoke ```focusParent()``` for every descendant? Or would it be possible to just
```js
child.hideChildren();
child.hideContainer();
```

Reply:
I agree that this would make sense and reduce overhead. I will test later that it doesn't cause issues in different circumstances such as closing all popups at once. Browsers seem to behave differently with focus. I just realized that master is also supposed to support Edge, but I don't have access to a Windows 10 machine right now.

Reply:
I can test on Edge, but honestly I wouldn't worry about making sure Edge works until the performance issues are ironed out with it. My initial commit was just API compatibility to make it so it runs without errors.

Reply:
Your suggestion worked on Chrome, but at least with the current implementation of `focusParent` it didn't work on Firefox. I don't know what exactly is going on with that. Thanks for helping with the testing!